src/lsm/forest.zig

```diff
@@ -133,6 +133,19 @@ pub fn ForestType(comptime Storage: type, comptime groove_config: anytype) type
             allocator.destroy(forest.node_pool);
         }
 
+        pub fn reset(forest: *Forest) void {
+            inline for (std.meta.fields(Grooves)) |field| {
+                @field(forest.grooves, field.name).reset();
+            }
```

Comment:
> Could this be forest.grooves.reset()? Similar to forest.node_pool.reset()?

Not without creating a new `Grooves` (collection-of-groove) abstraction, which is what the forest is anyway.
Right now `Grooves` is a
```zig
    const Grooves = @Type(.{
        .Struct = .{
            .layout = .Auto,
            .fields = groove_fields,
            .decls = &.{},
            .is_tuple = false,
        },
    });
```
..which cannot have methods. Iterating over the grooves with an `inline for` better matches our other functions (e.g. `deinit()`) too.

> And then both reset functions can return a pointer, so that we can chain and do .groove = forest.grooves.reset() below.

I don't like this, b/c it obfuscates the fact that it mutates the `forest.grooves` in-place.

> This also makes me wonder as the reader, who ultimately owns grid? A comment to explain why we don't reset the grid here would leave no doubt.

I'll add a comment explaining.